torch_xla_py/xla_dist.py

```diff
@@ -402,3 +408,249 @@ def get_cluster(self):
     cluster.validate()
     return cluster
 
+
+class DistributedExecutor(object):
+
+  SCRIPT_PATH_TMPL = '/tmp/{pid}/dist_training_ptxla_{worker}.sh'
+  MASTER_IDX = 0
+  MESH_SERVICE_PORT = 8477  # Use single port to disallow concurrent runs
+  DIST_ENV_VARS = [
+      'XRT_TPU_CONFIG', 'XRT_LOCAL_WORKER', 'XRT_MESH_SERVICE_ADDRESS'
+  ]
+
+  @staticmethod
+  def _parse_container_name(cmd):
+    cmd_str = ' '.join(cmd)
```

Comment:
What I meant was, iterate the single arguments in the list, and parse each one, instead of concatenating all of them into one string, and regex that.

Reply:
Arguments should be `--name=value`.
We do create those, no?

Reply:
The issue I have with space-concat and regex, is that we lose information.
What if an argument had a space, for example?
I think it is better to have an API which parses an arg (given its name) in an argument list.